src/pages/commits/CommitsWrapper.tsx

```diff
@@ -19,27 +26,40 @@ import {
   RenderCommitsBuildVariants,
 } from "./RenderCommit";
 
+const { parseQueryString, getString } = queryString;
 const { white } = uiColors;
 
+const DEFAULT_CHART_TYPE = ChartTypes.Absolute;
+
 interface Props {
   versions: Commits;
   error?: ApolloError;
   isLoading: boolean;
-  chartType?: ChartTypes;
   hasTaskFilter: boolean;
   hasFilters: boolean;
-  onChangeChartType: (chartType: ChartTypes) => void;
 }
 
 export const CommitsWrapper: React.FC<Props> = ({
   versions,
   isLoading,
   error,
-  chartType,
   hasTaskFilter,
   hasFilters,
-  onChangeChartType,
 }) => {
+  const { search } = useLocation();
+  const updateQueryParams = useUpdateURLQueryParams();
+  const parsed = parseQueryString(search);
+
+  const onChangeChartType = (chartType: ChartTypes): void => {
+    updateQueryParams({
+      [ChartToggleQueryParams.chartType]: chartType,
+    });
+  };
+
+  const chartType =
+    (getString(parsed[ChartToggleQueryParams.chartType]) as ChartTypes) ||
+    DEFAULT_CHART_TYPE;
+
```

Comment:
I think it would be possible to have this entire logic (toggling the chart, updating query params) in `ChartToggle.tsx`. This might be a slight improvement because then all of the logic would be localized to where the actual code is for the chart toggle

Reply:
yeah i'm not a big fan of this pattern either. I think extracting this type of logic into a hook reminiscent of `useState` would be the best approach. 
`useQueryState()` or something like that 
```
const [chartType, setChartType] = useQueryState(ChartToggleQueryParams.chartType, DEFAULT_CHART_TYPE);
```

How do you feel about some thing like that it could reduce allot of this boiler plate logic we use?

Reply:
it would operate in the same way as `useState` and trigger rerenders in the same way.

Reply:
I think it's a good idea and worth investigating in a separate ticket, it would definitely help if it can centralize most or all of the URL logic we copy/paste everywhere 😲